src/generic/generic-component-logo/generic-component-logo.twig

```diff
@@ -1,21 +1,52 @@
-{% set _class = "" %}
+{#
+  Parameters:
+    - "type" (string) (default: ''): "", "below", "right"
+    - "title" (string) (default: 'European Commission')
+    - "href" (string) (default: '')
+    - "extra_classes" (string) (default: '')
+    - "extra_attributes" (array) (default: []): format: [
+        {
+          "name" (string) (default: ''),
+          "value" (string) (default: '')
+        },
+        ...
+      ]
+#}
 
-{% if type is defined %}
-  {% if type == "below" %}
-    {% set _class = " ecl-logo--logotypebelow" %}
-  {% elseif type == "right" %}
-    {% set _class = " ecl-logo--logotype" %}
+{# Internal properties #}
+
+{% set _css_class = 'ecl-logo' %}
+{% set _extra_attributes = '' %}
+{% set _type = type|default('') %}
+{% set _title = title|default('European Commission') %}
+{% set _href = href|default('') %}
+
+{# Internal logic - Process properties #}
+
+{% if _type is not empty %}
+  {% if _type == 'below' %}
+    {% set _css_class = _css_class ~ ' ecl-logo--logotypebelow' %}
+  {% elseif _type == 'right' %}
+    {% set _css_class = _css_class ~ ' ecl-logo--logotype' %}
   {% endif %}
 {% endif %}
 
-{% if extra_classes%}
-  {% set _class = _class ~ " " ~ extra_classes%}
+{% if extra_classes is defined and extra_classes is not empty %}
```

Comment:
Default is missing for this one.

Reply:
what do you mean?

Reply:
In fact, seems `extra_classes` is documented to have a default, but it's not set as such, maybe it's me getting confused about this specific property?

Reply:
looks like you've just found a severe flaw in our normalization process 😀